cmd/gcp-controller-manager/node_annotator.go

```diff
@@ -234,24 +234,19 @@ type annotator struct {
 	annotate func(*core.Node, *compute.Instance) bool
 }
 
+var (
+	nodeURLRegexpStr = "^gce://([^/]+)/([^/]+)/([^/]+)$"
+	nodeURLRegexp    = regexp.MustCompile(nodeURLRegexpStr)
+)
+
 func parseNodeURL(nodeURL string) (project, zone, instance string, err error) {
-	u, err := url.Parse(nodeURL)
-	if err != nil {
-		return "", "", "", fmt.Errorf("failed to parse %q: %v", nodeURL, err)
-	}
-	if u.Scheme != "gce" {
-		return "", "", "", fmt.Errorf("instance %q doesn't run on gce", nodeURL)
-	}
-	project = u.Host
-	parts := strings.Split(u.Path, "/")
-	if len(parts) != 3 {
-		return "", "", "", fmt.Errorf("failed to parse %q: expected a three part path", u.Path)
-	}
-	if len(parts[0]) != 0 {
-		return "", "", "", fmt.Errorf("failed to parse %q: part one of path to have length 0", u.Path)
+	parts := nodeURLRegexp.FindStringSubmatch(nodeURL)
+	if len(parts) != 4 {
+		return "", "", "", fmt.Errorf("failed to parse %q: must match %q", nodeURL, nodeURLRegexpStr)
```

Comment:
`nodeURLRegexp.String()` will return the raw regexp string passed to MustCompile, so you don't need to store `nodeURLRegexpStr` separately.